reactive_kernel/kernel.py

```diff
@@ -22,15 +23,65 @@ def __init__(self, **kwargs):
 
     def do_execute(self, code, silent, store_history=True, user_expressions=None,
                    allow_stdin=False):
-        print("Executing code")
-        print(code)
+        
+        symbols = symtable(code, '<string>', 'exec')
+
         if not silent:
-            stream_content = {'name': 'stdout', 'text': code}
-            self.send_response(self.iopub_socket, 'stream', stream_content)
+            try: 
+                obj = Code_Object(code)
+            except Exception as e:
+                return {
+                    'status' : 'error',
+                    'ename' : e.expr,
+                    'evalue' : e.expr,
+                    'traceback' : sys.exc_info()
+                }
+            #stream_content = {'name': 'stdout', 'text': obj.input_vars}
+            #self.send_response(self.iopub_socket, 'stream', stream_content)
 
         return {'status': 'ok',
                 # The base class increments the execution count
                 'execution_count': self.execution_count,
                 'payload': [],
                 'user_expressions': {},
                 }
+
+
+class Code_Object:
+    def __init__(self, code):
+        self.input_vars = self.input_variables(code)
+        self.output_vars = self.output_variables(code)
+        self.code = code
+
+
+    def input_variables(self, code):
+        input_vars = []
+        symbols = symtable(code, '<string>', 'exec')
+        for i in symbols.get_symbols() :
+            if(i.is_global()) :
+                input_vars.append(i)
+        return str(input_vars)
+
+    def output_variables(self, code):
```

Comment:
Both the detection of input and output variables will not handle nested definitions or uses of variables.